src/dialogs/restart/dialog-restart-wait.ts

```diff
@@ -0,0 +1,175 @@
+class DialogRestartWait extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _open = false;
+
+  @state()
+  private _title = "";
+
+  private _actionOnIdle?: () => Promise<void>;
+
+  @state()
+  private _error?: string;
+
+  @state()
+  private _backupState?: ManagerState;
+
+  private _backupEventsSubscription?: UnsubscribeFunc;
+
+  @query("ha-md-dialog") private _dialog?: HaMdDialog;
+
+  public async showDialog(params: RestartWaitDialogParams): Promise<void> {
+    this._open = true;
+    this._loadBackupState();
+
+    this._title = params.title;
+    this._backupState = params.initialBackupState;
+
+    this._actionOnIdle = params.action;
+  }
+
+  private _dialogClosed(): void {
+    this._open = false;
+
+    if (this._backupEventsSubscription) {
+      this._backupEventsSubscription();
+      this._backupEventsSubscription = undefined;
+    }
+
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  public closeDialog(): void {
+    this._dialog?.close();
+  }
+
+  private _getWaitMessage() {
+    switch (this._backupState) {
+      case "create_backup":
+        return this.hass.localize("ui.dialogs.restart.wait_for_backup");
+      case "receive_backup":
+        return this.hass.localize("ui.dialogs.restart.wait_for_upload");
+      case "restore_backup":
+        return this.hass.localize("ui.dialogs.restart.wait_for_restore");
+      default:
+        return "";
+    }
+  }
+
+  protected render() {
+    if (!this._open) {
+      return nothing;
+    }
+
+    const waitMessage = this._getWaitMessage();
+
+    return html`
+      <ha-md-dialog
+        open
+        @closed=${this._dialogClosed}
+        .disableCancelAction=${true}
+      >
+        <ha-dialog-header slot="headline">
+          <ha-icon-button
+            slot="navigationIcon"
+            .label=${this.hass.localize("ui.common.cancel")}
+            .path=${mdiClose}
+            @click=${this.closeDialog}
+          ></ha-icon-button>
+          <span slot="title" .title=${this._title}> ${this._title} </span>
+        </ha-dialog-header>
+        <div slot="content" class="content">
+          ${this._error
+            ? html`<ha-alert alert-type="error"
+                >${this.hass.localize("ui.dialogs.restart.error_backup_state", {
+                  error: this._error,
+                })}</ha-alert
+              > `
+            : html`
+                <ha-spinner></ha-spinner>
+                ${waitMessage}
+              `}
+        </div>
+      </ha-md-dialog>
+    `;
+  }
+
+  private async _loadBackupState() {
+    try {
+      const { state: backupState } = await fetchBackupInfo(this.hass);
+      this._backupState = backupState;
+
+      if (this._backupState === "idle") {
+        this.closeDialog();
+        await this._actionOnIdle?.();
+        return;
+      }
+
+      this._backupEventsSubscription = await subscribeBackupEvents(
```

Comment:
You should not await here, if the dialog is closed before the promise is resolved we will never unsubscribe.